docs/developer-guide/ksqldb-clients/contributing.md

```diff
@@ -0,0 +1,175 @@
+
+### Creation and deletion of streams and tables ###
+
+These DDL/DML statements include:
+- Creation of source streams and tables: `CREATE STREAM`, `CREATE TABLE`
+- Creation of persistent queries: `CREATE STREAM AS SELECT`, `CREATE TABLE AS SELECT`
+- Deletion of streams and tables: `DROP STREAM`, `DROP TABLE`
+
+Send these DDL/DML requests to the [`/ksql` server endpoint](../ksqldb-rest-api/ksql-endpoint.md).
+
+As explained in the [server endpoint documentation](../ksqldb-rest-api/ksql-endpoint.md),
+the HTTP request body contains the text of the ksqlDB statement as well as any relevant
+query properties. Because ksqlDB supports many different types of statements for creating streams,
+tables, and persistent queries, the client interface should accept a SQL statement and pass the statement
+to the server endpoint directly.
+
+The server response will contain a command ID, command sequence number, and status information.
+These pieces of information are typically do not inform actions of ksqlDB users and do not need to be
+exposed as part of the client interface as a result.
+
+Note that the `/ksql` server endpoint accepts not only DDL/DML requests as described here, but also
+other types of statements including admin operation requests described [above](#listing-existing-streams-tables-topics-and-queries).
+If the server response for a SQL statement submitted to the `/ksql` endpoint has HTTP status code 200
+but the response body does not fit the form documented above, then the submitted SQL statement did
+not correspond to a DDL/DML statement. The client should handle this potential scenario by returning
+an error to the user.
+
+### Terminating persistent queries ###
+
+Send these requests to the [`/ksql` server endpoint](../ksqldb-rest-api/ksql-endpoint.md).
+
+The SQL statement that should be sent in the HTTP request body is `TERMINATE <persistent_query_id>;`.
+
+As noted in the [server endpoint documentation](../ksqldb-rest-api/ksql-endpoint.md),
+the format of the response body for terminate query requests is the same as for other DDL/DML requests
+[above](#creation-and-deletion-of-streams-and-tables).
+
+Configurations
+--------------
+
+The client must expose options for specifying the address of the ksqlDB server to connect to.
+Additional options that are important to expose include:
+- Support for TLS-enabled ksqlDB servers
+- Support for mutual-TLS-enabled ksqlDB servers
+- Support for ksqlDB servers with HTTP basic authentication enabled
+
+As an example, users specify options for the Java client via a [ClientOptions](api/io/confluent/ksql/api/client/ClientOptions.html)
+object that is passed when [creating a Client instance](api/io/confluent/ksql/api/client/Client.html#create(io.confluent.ksql.api.client.ClientOptions)).
+
+It may also be nice to support configuring multiple ksqlDB server addresses, so the client may route requests
```

Comment:
```suggestion
It is encouraged that clients support multiple ksqlDB server addresses, so the client may route requests
```

Not that our CLI does! (Unless that's changed recently)

Reply:
Does our CLI actually support this? I know the REST client (and therefore the CLI) supports parsing multiple server addresses, but AFAIA it only ever uses the first one (i.e., no fallback logic if the first one becomes unavailable): https://github.com/confluentinc/ksql/blob/master/ksqldb-rest-client/src/main/java/io/confluent/ksql/rest/client/KsqlRestClient.java#L110

Reply:
No... you obviously missed the last line of my comment.

Of course, the CLI _should_ support this ;)

Reply:
Oh LOL I read your comment as "Note that our CLI does" rather than "Not that our CLI does" -- I read it two or three times the same way each time :P Glad we're on the same page.